misk/src/main/kotlin/misk/security/authz/AccessAnnotation.kt

```diff
@@ -0,0 +1,45 @@
+package misk.security.authz
+
+import kotlin.reflect.KClass
+
+/**
+ * Use this to alias an annotation to a set of services and roles. This can be used to decouple code
+ * that needs access control from the policy that defines it.
+ *
+ * To demonstrate, let's define an access annotation. By convention these annotations are suffixed
+ * `Access`:
+ *
+ * ```
+ * @Retention(AnnotationRetention.RUNTIME)
+ * @Target(AnnotationTarget.FUNCTION)
+ * annotation class PaleontologistAccess
+ * ```
+ *
+ * Next we define actions that apply our annotation:
+ *
+ * ```
+ * class DiscoverDinosaurAction : WebAction {
+ *   @Get("/discover")
+ *   @PaleontologistAccess
+ *   fun discover()
+ * }
+ *
+ * class DigUpDinosaurAction : WebAction {
+ *   @Get("/dig")
+ *   @PaleontologistAccess
+ *   fun dig()
+ * }
+ * ```
+ *
+ * Finally we use multibindings to specify which services and roles are permitted:
+ *
+ * ```
+ * multibind<AccessAnnotation>().toInstance(AccessAnnotation(
+ *     DigUpDinosaurAction::class, roles = listOf("paleontologist", "intern")))
+ * ```
+ */
+data class AccessAnnotation(
+  val annotation: KClass<out Annotation>,
+  val services: List<String> = listOf(),
```

Comment:
I think ``AccessAnnotation`` is fine for roles since the mapping from concrete entity (a username) to the code facing entity (a role) is maintained externally. But I don't think we should be using code to describe the the inbound RBAC rules for peer services; we should be moving these to an externally managed system as well.

Reply:
Sure! Though I’m not quite sure what that looks like. One option is to bind the AccessAnnotation to something computed rather than something static. The Misk API stays the same mostly.